engine/src/card.cpp

```diff
@@ -1088,30 +1088,43 @@ void MCCard::timer(MCNameRef mptr, MCParameter *params)
 		MCObject::timer(mptr, params);
 }
 
+bool MCCard::isdeletable()
+{
+    bool t_deletable = true;
+    
+    if (t_deletable)
+        t_deletable = MCObject::isdeletable();
+    
+    if (t_deletable)
+        t_deletable = !getflag(F_C_CANT_DELETE);
```

Comment:
**In the IDE**

- **revFrontScript** 
    - `revDeleteStacks` should ensure all `cantDelete` properties are false and possibly use send in time to ensure the card doesn't have a script depth
- **revMenubar**
    - `revMenubarContextMenu` presents a contextual menu offering the following without checking the cantDelete:
        - remove mainstack from memory
        - delete substack
        - delete card
        - delete group
- **revBackscriptLibrary**
    - `reloadStack` (should be in front script?) set all `cantDelete` properties to false and use send in time to clear the script depth from the stack sent the `reloadStack` message. There also appears to be some unnecessary code duplication in there also so sort that out. _Stack name is already loaded hell_ syndrome is a horrible seemingly impossible to get out of recursive loop that people often complain about. I think I can see why it happens here.
    - `__ideRemoveFromMemory` check the `cantDelete` and deal gracefully with errors
    - `revIDEDeleteObjects`  check the `cantDelete` and deal gracefully with errors
    - `revIDEDeleteCard`   check the `cantDelete` and deal gracefully with errors
    - `revCheckGroupDelete` probably should check the `cantDelete` of the group and put up dialog
    - `revIDEUpdatePlugins` should ensure all `cantDelete` properties are false
    - `revIDEMoveCard`should ensure all `cantDelete` properties are false and reset after to original settings and possibly use send in time to ensure the card doesn't have a script depth.
    - `revIDEMoveControl`should ensure all `cantDelete` properties are false and reset after to original settings and possibly use send in time to ensure the card doesn't have a script depth. Also there's two implementations so we should delete one!!!!
    - `revIDEDeleteObjects` should ensure all `cantDelete` properties are false and reset after to original settings and possibly use send in time to ensure the card doesn't have a script depth
- **revSaveAsStandalone**
    - `revStandaloneDeployWithParams` should ensure all `cantDelete` properties are false and possibly use send in time to ensure the card doesn't have a script depth
    - `revCopyExternals` should ensure all `cantDelete` properties are false and possibly use send in time to ensure the card doesn't have a script depth
    - `revCleanBuildStacks` should ensure all `cantDelete` properties are false and possibly use send in time to ensure the card doesn't have a script depth
    - `revSaveStackFile` should ensure all `cantDelete` properties are false and possibly use send in time to ensure the card doesn't have a script depth
    - `revCloseOpenStacks` should ensure all `cantDelete` properties are false and possibly use send in time to ensure the card doesn't have a script depth
    - `revSBResaveScriptOnlyStackAsProperStackFile` should ensure all `cantDelete` properties are false and possibly use send in time to ensure the card doesn't have a script depth
- **Script Editor** - I got an error from the script editor and I suspect it is because something is being deleted script depth with a non-0 script depth. It needs investigation but here is where getting an error with the actual object causing the problem would be very helpful!

**In the engine:**

- The `revert` command behavior with `this stack` from within the scripts of the stack is completely broken by this PR (I think quite reasonably!). Previously it may have worked if the command was called from a script of the stack other than the stack script. However, this seems very unsafe to me and I think we should kill it! With fire 🔥 

For example:

```
-- button script
on mouseUp
   revert
end mouseUp
```

- Perhaps we need a variation on `delete <object>`... `delete <object> with force`? Would ignore cantDelete flags but not script depth and other dangerous things. This would remove some code from the IDE and avoid having to add a recursive handler that ensured an object didn't have any cantDelete props that would block delete.